app/controllers/users_controller.rb

```diff
@@ -1,4 +1,13 @@
 class UsersController < ApplicationController
+  before_action :find_user, only: [:show, :edit, :update, :destroy]
+  before_action :logged_in_user, only: [:index, :edit, :update]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
+
+  def index
+    @users = User.all.page(params[:page]).per(10)
```

Comment:
cơ chế của kaminari là gì nhỉ? (load hết dữ liệu ra rồi phân trang hay chỉ load theo số lượng bản ghi được khai báo)

Reply:
một trong những mục đích của việc phân trang là để tránh load một lúc một lượng dữ liệu lớn, vậy nên em nghĩ cơ chế của nó là load theo số lượng bản ghi được thiết lập ạ (^^;)

Reply:
nghĩ thôi á? chắc chắn xem nào em? lấy bằng chứng đi =))

Reply:
em chắc chắn luôn :1st_place_medal:  vì đợi request trên 1 trang web csdl có 10000 bản ghi không phân trang sẽ chậm hơn so với có phân trang :v: em tự tạo seed để thử luôn rồi ạ (^^;)

Reply:
put number 10 to config file